File: kaushik/kafka_py/topic.py
from kafka import KafkaAdminClient
from kafka.admin import NewTopic

def main():
    try: 
        print("Attempting to Connect...!")
        admin = KafkaAdminClient(
            bootstrap_servers= ["localhost:9093", "localhost:9094", "localhost:9095"]
        )
        print("Connected!") 

        cars = NewTopic(name='cars', num_partitions=2, replication_factor=3)
        admin.create_topics(new_topics=[cars])
        print("Topics created!")
    except Exception as e:
        print("Something went wrong:" + e)

if __name__ == "__main__":
    main()

File: kaushik/kafka_js/topic.js
const {Kafka} = require("kafkajs")

run();

async function run() {
    try {
        const kafka = new Kafka({
            "clientId": "myapp",
            "brokers": ["localhost:9093", "localhost:9094", "localhost:9095"]
        })

        const admin = kafka.admin();
        console.log("Waiting to Connect...!")
        await admin.connect //promise, therefore, we need AWAIT
        console.log("Connected!")

        await admin.createTopics({
            "topics": [{
                "topic": "Users",
                "numPartitions": 2 
            }]
        })
        console.log("Topics created successfully!")
        await admin.disconnect();
    } catch (error) {
        console.error(`Something bad happened ${error}`)
    }
    finally{
        process.exit(0);
    }
}